src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private static final Object[] DEFAULT_EMPTY_DATA_ARRAY = {};
+    private static final String INVALID_INDEX_ERROR_MESSAGE = 
+            "Index should be bigger then 0 and less then ArrayList size";
```

Comment:
The error message for the index validation is not informative enough. It's better to include the actual index and the size of the ArrayList in the message so that it's clearer what went wrong.